Svc/CmdDispatcher/CommandDispatcherImpl.cpp

```diff
@@ -32,6 +32,19 @@ namespace Svc {
     }
 
     void CommandDispatcherImpl::compCmdReg_handler(NATIVE_INT_TYPE portNum, FwOpcodeType opCode) {
+
```

Comment:
Couldn't this be done in one step by adding and elseif on line 44:

```c++
else if (this->m_entryTable[slot].used and (opCode == this->m_entryTable[slot].opcode) and ALLOW_REREGISTRATION) {
                this->m_entryTable[slot].opcode = opCode;
                this->m_entryTable[slot].port = portNum;
                this->m_entryTable[slot].used = true;
                this->log_DIAGNOSTIC_OpCodeRegistered(opCode,portNum,slot);
                slotFound = true;
}
```

I believe your method should work as well, it just splits the process into 2 halves.

Reply:
You might also consider making the above emit a "OpCodeReregistered" EVR just so it is clear that the opcode is being reused.

Reply:
Added `OpCodeReregistered` EVR on CMD reregistration.